src/ucp/api/ucp_def.h

```diff
@@ -771,5 +772,22 @@ typedef struct ucp_ep_params {
  */
 #define UCP_ENTITY_NAME_MAX 32
 
+/**
+ * @ingroup UCP_TRANSPORT
+ * @brief Maximum number of transport/device name pairs
+ */
+#define UCP_MAX_TRANSPORTS 10
+
+/**
+ * @ingroup UCP_ENDPOINT
+ * @brief Array of transport layers and device names used by an endpoint
+ */
+typedef struct ucp_transports {
+    unsigned int num_entries;
+    struct {
+        char transport_name[UCT_TL_NAME_MAX];
+        char device_name[UCT_DEVICE_NAME_MAX];
```

Comment:
maybe just return "const char *" from UCP, and these strings will be considered valid as long as the endpoint is valid? Or define other constant for max length? To avoid including uct.h from ucp.h

Reply:
Since the endpoint should be valid at the time I issue the queries, I changed the structures to use const char * which resolves my problems with using UCT headers and defining string sizes.